flytekit/types/structured/arrow.py

```diff
@@ -0,0 +1,68 @@
+import os
+import re
+import typing
+from abc import ABC
+
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+from google.cloud import bigquery, bigquery_storage
+from google.cloud.bigquery import LoadJob, LoadJobConfig
+from google.cloud.bigquery_storage_v1 import types
+
+from flytekit.types.structured.structured_dataset import (
+    FLYTE_DATASET_TRANSFORMER,
+    DatasetFormat,
+    DatasetPersistenceHandler,
+    DatasetRetrievalHandler,
+)
+
+
+class ArrowToParquetPersistenceHandlers(DatasetPersistenceHandler, ABC):
+    def persist(
+        self,
+        table: pa.Table,
+        to_file: os.PathLike,
+    ):
+        pq.write_table(table, to_file)
+
+
+class ArrowToBQPersistenceHandlers(DatasetPersistenceHandler, ABC):
+    def persist(self, table: pa.Table, path: str, job_config: typing.Optional[LoadJobConfig] = None) -> LoadJob:
+        table_id = path.split("://", 1)[1].replace(":", ".")
+        client = bigquery.Client()
+        return client.load_table_from_dataframe(table.to_pandas(), table_id, job_config=job_config)
+
+
+class ParquetToArrowRetrievalHandler(DatasetRetrievalHandler, ABC):
+    def retrieve(self, path: str) -> pa.Table:
+        return pq.read_table(path)
+
+
+class BQToArrowRetrievalHandler(DatasetRetrievalHandler, ABC):
```

Comment:
after we clean this up a bit more and have a working example, let's send this to dylan and see what he thinks.